ml-agents-envs/mlagents/envs/environment.py

```diff
@@ -625,6 +627,7 @@ def _get_state(self, output: UnityRLOutput) -> Tuple[AllBrainInfo, bool]:
             )
         return _data, global_done
 
+    @timed
     def _generate_step_input(
```

Comment:
What is the purpose of timing this in addition to timing the env.step time here?

Reply:
I put it there because that's where the protobuf serialization occurs. I'll put one on RpcCommunicator.communicate too.